app/services/resolve_remote_account_service.rb

```diff
@@ -11,97 +11,153 @@ class ResolveRemoteAccountService < BaseService
   # @param [String] uri User URI in the form of username@domain
   # @return [Account]
   def call(uri, update_profile = true, redirected = nil)
-    username, domain = uri.split('@')
+    @username, @domain = uri.split('@')
 
-    return Account.find_local(username) if TagManager.instance.local_domain?(domain)
+    return Account.find_local(@username) if TagManager.instance.local_domain?(@domain)
 
-    account = Account.find_remote(username, domain)
-    return account unless account_needs_webfinger_update?(account)
+    @account = Account.find_remote(@username, @domain)
+
+    return @account unless webfinger_update_due?
 
     Rails.logger.debug "Looking up webfinger for #{uri}"
 
-    data = Goldfinger.finger("acct:#{uri}")
+    @webfinger = Goldfinger.finger("acct:#{uri}")
 
-    raise Goldfinger::Error, 'Missing resource links' if data.link('http://schemas.google.com/g/2010#updates-from').nil? || data.link('salmon').nil? || data.link('http://webfinger.net/rel/profile-page').nil? || data.link('magic-public-key').nil?
+    raise Goldfinger::Error, 'Missing resource links' if links_missing?
```

Comment:
The only reason this was made to raise this error is so we could display feedback to the user. But I am seeing sidekiq jobs failing *and retrying* with that error, which should not happen, because there's no point in retrying from this error. I haven't changed it because I didn't want to change functionality in the same PR, but this needs to be changed.